### PR TITLE
[Issue tracker] - fix can't Delete attachment #8006

### PR DESCRIPTION
**Pull request by kongtiaowang**
[Issue tracker] - Delete attachment #8006
A clear and concise description of what the bug is
If a user does not have Issue Tracker: Close/Edit/Re-assign/Comment on All Issues, they cannot delete their own attachment.
#### Testing instructions 
If a user does not have Issue Tracker: Close/Edit/Re-assign/Comment on All Issues, they can delete their own attachment.

### CODE REVIEW

**Comment by ridz1208** [comment]
@kongtiaowang can you rebase this to go to 24.1-release please

**Comment by ridz1208** [comment]
@GeorgeMurad could you please test this so we merge it into the next bugfix ?

**Comment by GeorgeMurad** [comment]
@ridz1208 I will!.

File: modules/issue_tracker/jsx/attachments/attachmentsList.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ class AttachmentsList extends Component {
    */
   displayAttachmentOptions(deleteData, item) {
     if (this.props.userHasPermission
-      || this.state.attachments.whoami === item.user) {
+      || this.state.issueData.whoami === item.user) {
```
Comment (zaliqarosli) [inline modules/issue_tracker/jsx/attachments/attachmentsList.js:103]: issueData isn't a defined state in this class. not sure what this is supposed to be?

**Review by zaliqarosli — CHANGES_REQUESTED** [review]
this.state.issueData doesn't exist

**Comment by zaliqarosli** [comment]
@GeorgeMurad any chance you could re-test this? thanks!

**Comment by zaliqarosli** [comment]
@kongtiaowang i feel like i should review this cz i made changes 😅 @driusan anyone who should/can review this? @laemtl ? it's quite quick

**Comment by laemtl** [comment]
@zaliqarosli  If this passes @GeorgeMurad tests maybe he can approves it?

**Comment by GeorgeMurad** [comment]
@laemtl Yes, I tested  it and it passes.

**Pull request merged**